wire-ios-data-model/Tests/Source/Model/Conversation/ZMConversationTests+SecurityLevel.swift

```diff
@@ -728,6 +728,44 @@ class ZMConversationTests_SecurityLevel: ZMConversationTestsBase {
         }
     }
 
+    func testItMarksMLSConversationAsNotVerifiedAfterResendMessage() async throws {
+        // GIVEN
+        let conversation = await syncMOC.perform {
+            let conversation = ZMConversation.insertNewObject(in: self.syncMOC)
+            conversation.conversationType = .group
+            conversation.messageProtocol = .mls
+            conversation.mlsVerificationStatus = .degraded
+
+            return conversation
+        }
+
+        let message = try await syncMOC.perform {
+            return try XCTUnwrap(
+                try conversation.appendText(content: "foo") as? ZMOTRMessage
+            )
+        }
+
+        await syncMOC.perform {
+            XCTAssertEqual(message.deliveryState, .failedToSend)
+            XCTAssertTrue(message.isExpired)
+            XCTAssertTrue(message.causedSecurityLevelDegradation)
+        }
```

Comment:
nitpick: this could be done in a single `syncMOC.perform` block to avoid switching between the async thread and the context's thread